transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollSocketChannelConfig.java

```diff
@@ -632,4 +634,9 @@ public EpollSocketChannelConfig setEpollMode(EpollMode mode) {
         super.setEpollMode(mode);
         return this;
     }
+
+    private void calculateMaxBytesPerGatheringWrite() {
+        // Multiply by 2 to give some extra space in case the OS can process write data faster than we can provide.
+        setMaxBytesPerGatheringWrite(getSendBufferSize() << 1);
```

Comment:
I think this is not very likely at all but should we guard against overflow ?

Reply:
agreed this is very unlikely, but shouldn't hurt.